slides/week0/week0.org

```diff
@@ -0,0 +1,484 @@
+
+* Variables
+- =my_var= now represents the value 2
+#+BEGIN_SRC C++ -n :exports both :includes <iostream> :defines cout std::cout
+  int main() {
+    int my_var = 2;
+    cout << my_var + my_var;
+    return 0;
+  }
+#+END_SRC
+
+* Declaration vs Definition
+- int num = 1;
+    - creates the variable num of type int and assigns it the value 1
+- num = 4;
+    - the type is already known from above so this line just assigns it the value 4
+
+* Variables
+- Variables can have their assigned value changed on the fly
+#+BEGIN_SRC C++ -n :exports both :includes <iostream> :defines cout std::cout
+  int main() {
+    int my_var = 2;
+    cout << my_var + my_var;
+    my_var = 4;
+    cout << my_var + my_var;
+    return 0;
+  }
+#+END_SRC
+
+* Questions??
+#+BEGIN_NOTES
+Write a couple examples of setting up a primitive already described and then
+changing it using only the operators above. like printing numbers 1 - 10 using a var.
+DO NOT use strings, ++, --, or anything else not covered already
+#+END_NOTES
+- Still more to come...
+- Let's write some code
+
+* If Statements
+- What *IF* I only want to execute code sometimes?
+
+* If Statement Syntax
+#+BEGIN_SRC C++ -n
+  int main() {
+    if (true) {
+        // <==== This executes =====
+    }
+    if (false) {
+        // <==== This does *NOT* execute =====
+    }
+  }
+#+END_SRC
+
+* Equivalence operators
+| symbol | True Condition | example true | example false
+|-------+------|------+------|
+| =!= | opposite of current value| !true | !false
+| ==== | "equal"| 2 == 2 | 4 == 2
+| =!== | *NOT* "equal"| 2 != 4 | 2 != 2
+| =>= | the number of the left is larger and not equal to the right| 4 > 2 | 2 > 4
+| =<= | the number of the left is smaller and not equal to the right| 2 < 4 | 4 < 2
+
+* If else syntax
+#+BEGIN_SRC C++ -n
+  int main() {
+    if (true) {
+        // <==== This executes =====
+    } else {
+        // <==== This does *NOT* execute =====
+    }
+    if (false) {
+        // <==== This does *NOT* execute =====
+    } else {
+        // <==== This executes =====
+    }
+  }
+#+END_SRC
+
+* Conditional Practice 1 Question
+#+BEGIN_NOTES
+1
+#+END_NOTES
+- 1 or 2?
+#+BEGIN_SRC C++ -n
+if (2 == 2) {
+    // 1
+} else {
+    // 2
+}
+#+END_SRC
+
+* Conditional Practice 2 Question
+#+BEGIN_NOTES
+2
+#+END_NOTES
+- 1 or 2?
+#+BEGIN_SRC C++ -n
+if (2 != 2) {
+    // 1
+} else {
+    // 2
+}
+#+END_SRC
+
+* Conditional Practice 3 Question
+#+BEGIN_NOTES
+2
+#+END_NOTES
+- 1 or 2?
+#+BEGIN_SRC C++ -n
+if (2 < 2) {
+    // 1
+} else {
+    // 2
+}
+#+END_SRC
+
+* Chaining Conditionals
+#+BEGIN_NOTES
+Talk about how this is difficult to write at deeper levels
+Think 5-6 Conditionals
+#+END_NOTES
+#+BEGIN_SRC C++ -n
+  int main() {
+    if (true) {
+        // <==== This executes =====
+        if (true) {
+            // <==== This executes =====
+        }
+        // <==== This executes =====
+    }
+    // <==== This executes =====
+  }
+#+END_SRC
+
+* Logical Operators *AND*
+- *&&*
```

Comment:
I don't like how these slides are laid out - maybe put the operator symbol in the table instead of `AND`?